### PR TITLE
System.Data.* missed Equals nullable annotations 

### PR DESCRIPTION
**Pull request by hrrrrustic**

### CODE REVIEW

File: src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs [hunk 1]
```diff
@@ -30,7 +32,7 @@ internal static DbConnectionPoolIdentity GetCurrent()
             throw new PlatformNotSupportedException();
         }
 
-        public override bool Equals(object? value)
+        public override bool Equals([NotNullWhen(true)] object? value)
```
Comment (roji) [inline src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs:35]: This doesn't hold when `this == NoIdentity` - the method will return true even if value is false. That definitely seems like questionable behavior, but I'm not sure there's an actual bug here and this old legacy code (AFAIK only used as a support type for the System.Data.{Odbc,OleDb}).

**Review by roji — CHANGES_REQUESTED** [review]
Continuation of #47598, /cc @stephentoub.

As in #49428, it would have been better if this attribute were implicitly inherited when specified on a base implementation...

File: src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs (hunk 1 above)
Comment (hrrrrustic) [reply src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs]: Reverted

File: src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs [hunk 2]
```diff
@@ -1,6 +1,8 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
```
Comment (roji) [inline src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs:4]: Can you please remove?

**Review by roji — CHANGES_REQUESTED** [review]

File: src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs (hunk 2 above)
Comment (hrrrrustic) [reply src/libraries/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbConnectionPoolIdentity.cs:4]: ugh, forgot about it, wait a second

**Review by roji — APPROVED** [review]

**Comment by roji** [comment]
Thanks @hrrrrustic

**Pull request merged**